contracts/interfaces/rysk/IRyskLiquidityPool.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * Describes the interface used to communicate with the Rysk LiquidityPool contract.
+ *
+ */
+interface IRyskLiquidityPool is IERC20 {
+    /**
+     * Gets the current deposit epoch.
+     *
+     * @notice Deposit epochs represent time resolution on which the pool handles deposits.
+     * For every new deposit epoch a new deposit price per share is calculated.
+     * Deposits are initiated in one epoch and completed during epoch calculation by
+     * minting appropriate amout of pool shares for that deposit, calculated as amount/pps.
+     *
+     * @notice Expected duration for an epoch is around 1 week at the moment.
+     *
+     * @return the current deposit epoch
+     */
+    function depositEpoch() external view returns (uint256);
+
+    /**
+     * Gets the current withdrawal epoch.
+     *
+     * @notice Epochs represent time resolution on which the pool handles withdrawals.
+     * For every new withdrawal epoch a new withdrawal price per share is calculated.
+     * Withdrawals are initiated in one epoch and completed in another.
+     *
+     * @notice Expected duration for an epoch is around 1 week at the moment.
+     *
+     * @return the current withdrawal epoch
+     */
+    function withdrawalEpoch() external view returns (uint256);
+
+    /**
+     * Gets the price per share for the requrested deposit epoch.
+     *
+     * @param _epoch the deposit epoch
+     *
+     * @return the price per share at the requested deposit epoch
+     */
+    function depositEpochPricePerShare(uint256 _epoch)
+        external
+        view
+        returns (uint256);
+
+    /**
+     * Gets the price per share for the requrested withdrawal epoch.
+     *
+     * @param _epoch the withdrawal epoch
+     *
+     * @return the price per share at the requested withdrawal epoch
+     */
+    function withdrawalEpochPricePerShare(uint256 _epoch)
+        external
+        view
+        returns (uint256);
+
+    /**
+     * Deposits the specified amount of underlying currency into the pool.
+     *
+     * @notice The amount of underlying currency deposited is converted to shares.
+     * On sucessful deposit, deposit receipt is created containing only the amount deposited, no unredeemed shares.
+     * Only when a new deposit epoch is initiated, corresponding number of shares is minted but
+     * the ownership of the shares is not transferred to the caller, they remain owned by the pool.
```

Comment:
How come the ownership is not transferred? I'm confused about this one. I see you asking for the number of shares of our contract `ryskLqPool.balanceOf(address(this));`?